launch_testing/launch_testing/asserts/assert_output.py

```diff
@@ -16,6 +16,19 @@
 
 from ..util import resolveProcesses
 
+from osrf_pycommon.terminal_color import remove_ansi_escape_senquences
+
+
+def normalize_lineseps(lines):
+    """
+    Normalize and then return the given lines to all use `\n`.
+    """
+    lines = lines.replace(os.linesep, '\n')
+    # This happens (even on Linux and macOS) when capturing I/O from an
+    # emulated tty.
+    lines = lines.replace('\r\n', '\n')
```

Comment:
@wjwwood hmm, funky... Do we know why?

Reply:
Yeah, it uses `\r` to move the carriage return back to the beginning of the line. Most print functions emulate this behavior for you when you're connected to a tty and print a lone `\n`.

It's a quirk of these kinds of things, which I've mostly learned from experience. There's scarcely any documentation about these behaviors across operating systems and different configurations. So yeah, funky.